Library/Homebrew/cache_store.rb

```diff
@@ -13,14 +13,33 @@ class CacheStoreDatabase
   # @param  [Symbol] type
   # @yield  [CacheStoreDatabase] self
   def self.use(type)
-    database = CacheStoreDatabase.new(type)
-    return_value = yield(database)
-    database.close_if_open!
+    @db_type_reference_hash ||= {}
+    @db_type_reference_hash[type] ||= {}
+    type_ref = @db_type_reference_hash[type]
+
+    type_ref[:count] ||= 0
+    type_ref[:count]  += 1
+
+    type_ref[:db] ||= CacheStoreDatabase.new(type)
+
+    return_value = yield(type_ref[:db])
+    if type_ref[:count].positive?
+      type_ref[:count] -= 1
+    else
+      type_ref[:count] = 0
+    end
+
+    if type_ref[:count].zero?
+      type_ref[:db].write_if_dirty!
+      type_ref.delete(:db)
+    end
+
     return_value
   end
 
   # Sets a value in the underlying database (and creates it if necessary).
   def set(key, value)
+    dirty!
```

Comment:
[![Dirty!](https://img.youtube.com/vi/C4wBLUBa8YI/0.jpg)](https://youtu.be/C4wBLUBa8YI)
SCNR @MikeMcQuaid